### PR TITLE
Fixing that naming of v1.12.4+ of docker-engine packages also contain…

### PR DESCRIPTION
**Pull request by CheMical**
… name of linux distro.

### CODE REVIEW

**Comment by mzedeler** [comment]
Hi :-)

I'll take a look :-)

File: dsh [hunk 1]
```diff
@@ -27,7 +27,9 @@ else
         echo 'Building dsh image'
         DOCKER_VERSION=${OPT_DOCKER_VERSION-$(docker -v | awk '{print $3}' | tr -d ,)}
         PHUSION_VERSION=$(docker run --rm $BASE_IMAGE lsb_release -cs)
-        DOCKER_PACKAGE=${OPT_DOCKER_PACKAGE-docker-engine=$DOCKER_VERSION-0~$PHUSION_VERSION}
+        # Due to change in name of docker-engine packages from v1.12.4 which also contains name of distribution
+        PHUSION_DIST=$(dpkg --compare-versions 1.12.4 le $DOCKER_VERSION && docker run --rm phusion/baseimage lsb_release -is | awk '{print tolower($0)"-"}')
+        DOCKER_PACKAGE=${OPT_DOCKER_PACKAGE-docker-engine=$DOCKER_VERSION-0~$PHUSION_DIST$PHUSION_VERSION}
```
Comment (mzedeler) [inline dsh:32]: Rewrite to `$PHUSION_DIST-$PHUSION_VERSION` and remove the dash above.

File: dsh [hunk 2]
```diff
@@ -27,7 +27,9 @@ else
         echo 'Building dsh image'
         DOCKER_VERSION=${OPT_DOCKER_VERSION-$(docker -v | awk '{print $3}' | tr -d ,)}
         PHUSION_VERSION=$(docker run --rm $BASE_IMAGE lsb_release -cs)
-        DOCKER_PACKAGE=${OPT_DOCKER_PACKAGE-docker-engine=$DOCKER_VERSION-0~$PHUSION_VERSION}
+        # Due to change in name of docker-engine packages from v1.12.4 which also contains name of distribution
+        PHUSION_DIST=$(dpkg --compare-versions 1.12.4 le $DOCKER_VERSION && docker run --rm phusion/baseimage lsb_release -is | awk '{print tolower($0)"-"}')
```
Comment (mzedeler) [inline dsh]: What does this fallback to if the version is above 1.12.4?

**Review by mzedeler — CHANGES_REQUESTED** [review]

File: dsh (hunk 2 above)
Comment (CheMical) [reply dsh]: Default to the same as 1.12.4 (ubuntu-):
dpkg --compare-versions 1.12.4 le $DOCKER_VERSION
1.12.4 <= $DOCKER_VERSION

Defaults for veriosn 1.12.3 and below to the empty string.

File: dsh (hunk 1 above)
Comment (CheMical) [reply dsh:32]: That would be nice and short, but the version of docker-engine depends on the host.

From https://apt.dockerproject.org/repo/dists/ubuntu-xenial/main/filelist you get that docker-engine packages are named differently starting with version 1.12.4. So package with version 1.12.3 is named ~xenial_amd64.deb and package with version 1.12.4 and above is named ~ubuntu-xenial_amd64.deb

**Pull request merged**